pavelib/quality.py

```diff
@@ -0,0 +1,123 @@
+"""
+Check code quality
+"""
+import json
+import os
+import re
+from string import join
+from paver.easy import BuildFailure, call_task, cmdopts, needs, sh, task
+
+ROOT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+REPORTS_DIR = os.path.join(ROOT_DIR, 'reports')
+
+PACKAGES = [
+    'VEDA',
+    'VEDA_OS01',
+    'control',
+    'frontend',
+    'youtube_callback',
+    'scripts',
+]
+
+@task
+@cmdopts([
+    ("limit=", "l", "limit for number of acceptable violations"),
+])
+def run_pep8(options):  # pylint: disable=unused-argument
+    """
+    Run pep8 on system code.
+    Fail the task if any violations are found.
+    """
+    violations_limit = int(getattr(options, 'limit', -1))
+
+    sh('pep8 . | tee {report_dir}/pep8.report'.format(report_dir=REPORTS_DIR))
+
+    num_violations, violations_list = _pep8_violations(
+        "{report_dir}/pep8.report".format(report_dir=REPORTS_DIR)
+    )
+    violations_list = ''.join(violations_list)
+
+    violations_message = "Too many pep8 violations. Number of pep8 violations: {num_violations}. ".format(
+        num_violations=num_violations
+    )
+    violations_message += "The limit is {violations_limit}. ".format(violations_limit=violations_limit)
+    violations_message += "\n\nViolations:\n{violations_list}".format(violations_list=violations_list)
+
+    # Fail if number of violations is greater than the limit
+    if num_violations > violations_limit > -1:
+        raise BuildFailure(violations_message)
+
+
+def _pep8_violations(report_file):
```

Comment:
this can be rename to `_count_pep8_violations` so as to be consistent with pylint method